packages/vest/src/core/produce/collectFailureMessages.js

```diff
@@ -11,35 +15,28 @@ import useTestObjects from 'useTestObjects';
 const collectFailureMessages = (severity, options) => {
   const [testObjects] = useTestObjects();
   const { group, fieldName } = options || {};
-  const res = testObjects.reduce((collector, testObject) => {
-    if (group && testObject.groupName !== group) {
-      return collector;
-    }
-
-    if (fieldName && testObject.fieldName !== fieldName) {
-      return collector;
-    }
-
-    if (!testObject.failed) {
-      return collector;
-    }
-
-    if (isMatchingSeverityProfile(severity, testObject)) {
-      return collector;
-    }
-
-    collector[testObject.fieldName] = (
-      collector[testObject.fieldName] || []
-    ).concat(testObject.statement);
-
-    return collector;
-  }, {});
-
-  if (fieldName) {
-    return res[fieldName] || [];
-  } else {
-    return res;
-  }
+  const { stateRef } = context.use();
+  const ctxRef = { stateRef };
+
+  return cache(
+    [testObjects, severity, options], context.bind(ctxRef, () => {
+      const res = testObjects.reduce((collector, testObject) => {
+        if (group && testObject.groupName !== group) return collector;
+        if (fieldName && testObject.fieldName !== fieldName) return collector;
+        if (!testObject.failed) return collector;
+        if (isMatchingSeverityProfile(severity, testObject)) return collector;
+
+        collector[testObject.fieldName] = (
+          collector[testObject.fieldName] || []
+        ).concat(testObject.statement);
+
+        return collector;
+      }, {});
+
+      if (fieldName) return res[fieldName] || [];
+      return res;
+    })
```

Comment:
Are you sure we need to bind the state to the context here? From the looks of it we do not use the state in this function or in any of the functions being called by it. I may be wrong about this - let's have a quick run of the tests without it to see if it actually works as expected.

```suggestion
    [testObjects, severity, options], () => {
      const res = testObjects.reduce((collector, testObject) => {
        if (group && testObject.groupName !== group) return collector;
        if (fieldName && testObject.fieldName !== fieldName) return collector;
        if (!testObject.failed) return collector;
        if (isMatchingSeverityProfile(severity, testObject)) return collector;

        collector[testObject.fieldName] = (
          collector[testObject.fieldName] || []
        ).concat(testObject.statement);

        return collector;
      }, {});

      if (fieldName) return res[fieldName] || [];
      return res;
    }
```

Reply:
removed state bind. ran tests and works as expected